src/main/java/cz/smarteon/loxone/LoxoneIconRegistry.java

```diff
@@ -0,0 +1,68 @@
+package cz.smarteon.loxone;
+
+import cz.smarteon.loxone.message.ImageCommand;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.requireNonNull;
+
+public class LoxoneIconRegistry implements CommandRequestResponseListener<ByteBuffer> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LoxoneIconRegistry.class);
+
+    private final Map<String, ByteBuffer> iconMap = new HashMap<>();
+    private LoxoneWebSocket loxoneWebSocket;
+
+    private CountDownLatch iconLatch;
+
+    @Override
+    public void registerWebSocket(final @NotNull LoxoneWebSocket loxoneWebSocket) {
+        this.loxoneWebSocket = loxoneWebSocket;
+    }
+
+    @Nullable
+    public synchronized ByteBuffer getIcon(final @NotNull String iconUuid) {
+        requireNonNull(loxoneWebSocket, "LoxoneWebSocket should be available!");
+        if (iconMap.containsKey(iconUuid)) {
+            return iconMap.get(iconUuid);
+        } else {
+            iconLatch = new CountDownLatch(1);
+            loxoneWebSocket.sendCommand(ImageCommand.genericControlCommand(iconUuid));
+            try {
+                final int timeout = loxoneWebSocket.getAuthTimeoutSeconds() * loxoneWebSocket.getRetries() + 1;
```

Comment:
do not use auth timeout, also no need for any retries?? I would recommend to use very short timeout for this.

Reply:
well sorry, I wasn't very clear. No need to tweak retries.

Let's introduce possibly configurable separate timeout?

Or we can keep it like this for now - the more I am thinking about it I feel it would be better to add some blocking request-response method to LoxoneWebSocket instead... but I need more time to try out....

Reply:
https://github.com/Smarteon/loxone-java/issues/200

Reply:
Added a timeout for file retrieval with 1 second default timeout